integrationtest/backend/elasticsearch/src/test/java/org/hibernate/search/integrationtest/backend/elasticsearch/management/ElasticsearchSchemaAttributeValidationIT.java

```diff
@@ -0,0 +1,890 @@
+						.indexContext( INDEX_NAME )
+						.contextLiteral( SCHEMA_VALIDATION_CONTEXT )
+						.indexFieldContext( "myField" )
+						.failure( "Invalid value for attribute 'store'. Expected 'true', actual is 'null'" )
+						.build() );
+	}
+
+	@Test
+	public void attribute_nullValue_valid() {
+		elasticSearchClient.index( INDEX_NAME ).deleteAndCreate();
+		elasticSearchClient.index( INDEX_NAME ).type().putMapping(
+				"{"
+					+ "'dynamic': 'strict',"
+					+ "'properties': {"
+							+ "'myField': {"
+									+ "'type': 'integer',"
+									+ "'null_value': 739"
+							+ "}"
+					+ "}"
+				+ "}"
+		);
+
+		validateSchemaConfig()
+				.withIndex( INDEX_NAME, ctx -> {
+							IndexSchemaElement root = ctx.getSchemaElement();
+							root.field( "myField", f -> f.asInteger().indexNullAs( 739 ) ).toReference();
+						}
+				)
+				.setup();
+	}
+
+	@Test
+	public void attribute_nullValue_missed() {
+		elasticSearchClient.index( INDEX_NAME ).deleteAndCreate();
+		elasticSearchClient.index( INDEX_NAME ).type().putMapping(
+				"{"
+					+ "'dynamic': 'strict',"
+					+ "'properties': {"
+							+ "'myField': {"
+									+ "'type': 'integer'"
+							+ "}"
+					+ "}"
+				+ "}"
+		);
+
+		SubTest.expectException( () -> validateSchemaConfig()
+				.withIndex( INDEX_NAME, ctx -> {
+							IndexSchemaElement root = ctx.getSchemaElement();
+							root.field( "myField", f -> f.asInteger().indexNullAs( 739 ) ).toReference();
+						}
+				)
+				.setup() )
+				.assertThrown()
+				.isInstanceOf( Exception.class )
+				.hasMessageMatching( FailureReportUtils.buildFailureReportPattern()
+						.indexContext( INDEX_NAME )
+						.contextLiteral( SCHEMA_VALIDATION_CONTEXT )
+						.indexFieldContext( "myField" )
+						.failure( "Invalid value for attribute 'null_value'. Expected '739', actual is 'null'" )
+						.build() );
+	}
+
+	@Test
+	public void attribute_nullValue_invalid() {
+		elasticSearchClient.index( INDEX_NAME ).deleteAndCreate();
+		elasticSearchClient.index( INDEX_NAME ).type().putMapping(
+				"{"
+					+ "'dynamic': 'strict',"
+					+ "'properties': {"
+							+ "'myField': {"
+									+ "'type': 'integer',"
+									+ "'null_value': 777"
+							+ "}"
+					+ "}"
+				+ "}"
+		);
+
+		SubTest.expectException( () -> validateSchemaConfig()
+				.withIndex( INDEX_NAME, ctx -> {
+							IndexSchemaElement root = ctx.getSchemaElement();
+							root.field( "myField", f -> f.asInteger().indexNullAs( 739 ) ).toReference();
+						}
+				)
+				.setup() )
+				.assertThrown()
+				.isInstanceOf( Exception.class )
+				.hasMessageMatching( FailureReportUtils.buildFailureReportPattern()
+						.indexContext( INDEX_NAME )
+						.contextLiteral( SCHEMA_VALIDATION_CONTEXT )
+						.indexFieldContext( "myField" )
+						.failure( "Invalid value for attribute 'null_value'. Expected '739', actual is '777'" )
+						.build() );
+	}
+
+	@Test
+	public void attribute_fielddata_ignored() {
```

Comment:
If we don't use fielddata, maybe it should just be removed from the model entirely, like you did for boost.
It only made sense when we allowed sorts on analyzed string fields, but we don't allow that anymore.